esmvaltool/utils/cmorize_obs.py

```diff
@@ -0,0 +1,273 @@
+        filemode='a',
+        format=out_fmt,
+        datefmt='%H:%M:%S',
+        level=config_user['log_level'].upper())
+    root_logger.setLevel(config_user['log_level'].upper())
+    logfmt = logging.Formatter(out_fmt)
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(logfmt)
+    root_logger.addHandler(console_handler)
+
+    # print header
+    logger.info(HEADER)
+
+    # run
+    timestamp1 = datetime.datetime.utcnow()
+    timestamp_format = "%Y-%m-%d %H:%M:%S"
+
+    logger.info(
+        "Starting the CMORization Tool at time: %s UTC",
+        timestamp1.strftime(timestamp_format))
+
+    logger.info(70 * "-")
+    logger.info("input_dir  = %s", config_user["rootpath"]["RAWOBS"][0])
+    logger.info("output_dir = %s", config_user["output_dir"])
+    logger.info(70 * "-")
+
+    # call the reformat function
+    if args.obs_list_cmorize:
+        obs_list = args.obs_list_cmorize
+    else:
+        obs_list = []
+    _cmor_reformat(config_user, obs_list)
+
+    # End time timing
+    timestamp2 = datetime.datetime.utcnow()
+    logger.info(
+        "Ending the CMORization Tool at time: %s UTC",
+        timestamp2.strftime(timestamp_format))
+    logger.info(
+        "Time for running the CMORization scripts was: %s",
+        timestamp2 - timestamp1)
+
+
+def _cmor_reformat(config, obs_list):
+    """Run the cmorization routine."""
+    logger.info("Running the CMORization scripts.")
+
+    # master directory
+    raw_obs = config["rootpath"]["RAWOBS"][0]
+
+    # set the reformat scripts dir
+    reformat_scripts = os.path.join(os.path.dirname(__file__),
+                                    '../cmor/cmorizers/obs')
+    run_dir = os.path.join(config['output_dir'], 'run')
+    # datsets dictionary of Tier keys
+    datasets = _assemble_datasets(raw_obs, obs_list)
+    logger.info("Processing datasets %s", datasets)
+
+    # loop through tier/datasets to be cmorized
+    for tier, _ in datasets.items():
+        for dataset in datasets[tier]:
+            reformat_script_root = os.path.join(reformat_scripts,
+                                                'cmorize_obs_' + dataset)
+            # in-data dir; build out-dir tree
+            in_data_dir = os.path.join(raw_obs, tier, dataset)
+            out_data_dir = os.path.join(config['output_dir'], tier, dataset)
+            if not os.path.isdir(out_data_dir):
+                os.makedirs(out_data_dir)
+
+            # all operations are done in the working dir now
+            os.chdir(out_data_dir)
+
+            # figure out what language the script is in
+            if os.path.isfile(reformat_script_root + '.ncl'):
+                reformat_script = reformat_script_root + '.ncl'
+                logger.info("CMORizing dataset %s using NCL script %s",
+                            dataset, reformat_script)
+                # copy over the reformat script
+                shutil.copy2(reformat_script, out_data_dir)
```

Comment:
Why are you copying and then running? This may lead non-reproducible results if people start editing those files and re-running. Best to just run from the installed esmvaltool.

Reply:
Just provide an absolute path to the script in the `Popen` command and it should work. If you need to load additional NCL scripts from the main NCL script, you can store the location of the reformat script directory in an environmental variable, e.g. reformat_scripts=/some/path/esmvaltool/cmor/cmorize and use `load "$reformat_scripts/obs/some_reformat_script.ncl"` in NCL for loading.

Reply:
le fixed